s2a_e2e_test.go

```diff
@@ -405,3 +410,101 @@ func TestNewTLSClientConfigFactoryWithoutTokenManager(t *testing.T) {
 		t.Errorf("tls.Config has unexpected certificate: got: %v, want: %v", got, want)
 	}
 }
+
+// startHTTPServer runs an HTTP server on a random local port and serves a /hello endpoint.
```

Comment:
Is there a reason why we only have HTTP tests at the top layer and only have GRPC tests in the v2 layer, rather than having both kinds of tests at both layers?

Reply:
I added the tests to both layers fro grpc. For http case, it doesn't call into internal code, that's why the test is only at the top level.